datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -2494,4 +2533,19 @@ mod tests {
         assert_eq!(result.as_ref(), &expected);
         Ok(())
     }
+
+    #[test]
+    fn bitwise_shift_scalar_test() -> Result<()> {
+        let input = Arc::new(Int32Array::from(vec![Some(2), None, Some(4)])) as ArrayRef;
+        let module = ScalarValue::from(10i32);
```

Comment:
Likewise, here a test for null handling might be good

Reply:
Instead of doing unit test here, I did a simple SQL test which test the whole pipeline, and I found that DF doesn't handle nulls for bitwise operators.

Fixed in:

https://github.com/apache/arrow-datafusion/pull/3037/commits/810f4b052c863a6a9d992a6eba1fd0e40f2c49d4